### DISCUSSION TITLE
Recording Audio and Video Locally

### DISCUSSION THREAD

**Discussion by MIDIculous** [ant-media/Ant-Media-Server#5686] [FAQs]
Is there a way to be able for the user to record the Audio and Video locally without going through a server?

**Comment by USAMAWIZARD** [comment]
Can you please let me know which SDK are you using Web / IOS / Android

**Comment by MIDIculous** [comment]
> Can you please let me know which SDK are you using Web / IOS / Android

We are using the Web SDK:

We want peer-to-peer audio + video + data channel, and we want to:
- Record the video locally while using P2P, without streaming it over a server
- Choose the audio output for the received audio track(s) in the stream
- Check which audio inputs and video sources are currently selected/streaming.

We're using WKWebView (WebKit) which doesn't have the HTMLMediaElement:captureStream() method. Is there still some way to do this?

**Reply by USAMAWIZARD** [reply]
I have asked this internally and I will get back to you soon

**Reply by USAMAWIZARD** [reply]
Currently we do not have a solution to record p2p call , as the video does not gets relay through the server , we do not support client side p2p call recording but  You can create a paid development request on the Git Hub discussion  may be someone can develop it for you.

**Comment by lastpeony** [comment]
You can make a 2 person conference and record the streams with custom html page + media push plugin or MCU

There is no client side recording solution provided by antmedia. To be honest client side recording is not a good idea.

**Reply by mekya** [reply]
Hi @MIDIculous, 
I agree that client-side recording is not a good idea. It's likely that you can lose all recording due to client-side issues.

I think you can capture local and remote video on the client side and push it to the server to be recorded at the same time.
I think it's an easy, fast, and reliable solution.  If you're interested, please let us know and we can help you

Regards
